src/main/java/com/amazon/dlic/util/RolesUtil.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.dlic.util;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RolesUtil {
+    private static final Logger log = LogManager.getLogger(RolesUtil.class);
+
+    /*
+     * Split the given rolesObject to one or several roles
+     * We accept String or JSONarray only
+     *
+     * rolesObject should be String if given by roles_key
+     * rolesObject should be String or ArrayList if given by roles_path
+     *
+     * Here are some examples:
+     *
+     *   JWT paylaoad contains:
+     *     "access": {
+     *       "roles": [
+     *         "readall, testrole",
+     *         "admin",
+     *         123,
+     *         "kibanauser"
+     *       ]
+     *     }
+     *  if contains a Json array, it should return elements that is a String.
+     *    roles_path = $["access"]["roles"]
+     *    should return ["readall, testrole", "admin", "kibanauser"]
+     *
+     *  if contains a String, it will split by comma to get roles.
+     *    roles_path = $["access"]["roles"][0]
+     *    should return ["readall", "testrole"]
+     *
+     *  if contains neither Json array nor String, e.g. Json Object, it should throw exception.
+     *    roles_path = $["access"]
+     *    should throw ElasticsearchSecurityException
```

Comment:
`ElasticsearchSecurityException` does not sound right to me in this case.

Reply:
This should be IllegalArgumentException, will edit